### DISCUSSION TITLE
Interceptors capabilities to change rows

### DISCUSSION THREAD

**Discussion by jromerus** [Mimetis/Dotmim.Sync#863] [Q&A]
Hi, regarding to interceptors I'm reading this:

> Or even intercept all the rows that are going to be applied on a destination database

I wonder if I could remove columns from specific rows. My aim is avoid syncing all the fields of a row, if possible. My software can track field changes and saves information of last field date time change from each row of tracked entities. With an intercerptor I could get which fields from that row need to be synced. So the question is, could I remove not changed columns and sync to the client (or to the server) only desired ones?

**Comment by Mimetis** [comment]
I don't think so, because DMS uses the order of the columns and the index to generate the statements.
That means if you don't have the same columns count / order than what is specified in the SyncSetup, it will fail.
That being said, I've never tested it...

Maybe, you can try different interceptors I guess, the nearest interceptors from the insert / update / delete command are, in order : `OnGetCommand `, `OnRowsChangesApplying` then `OnExecuteCommand`.

**Reply by jromerus** [reply]
Ok, I thought column name was included in the data sended or received. Do you think I could add that information into the "row package", in order to send only desired columns with names (changed columns) and process them overriding standard behaviour? I suppose I would need to override serialization, deserialization and database command building based on new deseraization that would include column name.

Instead of using my tracking column changes maybe I could use a a more "universal" approach with interceptores saving a "snap shot" of each processed row into some kind of cache (with string to hash to reduce size) , and analyze row differences (actual row against previous one).

**Reply by Mimetis** [reply]
Yes, including column names is a feature I wanted to integrate, but I need to stay compatible with the previous versions as much as I possibly can.
Maybe in a future version, I guess.

Before sending the generated batch files, using [OnTableChangesSelected](https://dotmimsync.readthedocs.io/Interceptors.html#ontablechangesselected) or even [OnDatabaseChangesSelected](https://dotmimsync.readthedocs.io/Interceptors.html#ondatabasechangesselected), you still open these batch files and add your custom json

Or, you can use the property `AdditionalProperties` from your `SyncContext` instance, to add anything you want, and use it on purpose on the other side.

Once again, hard to help you here, without a simple example to reproduce.
